File: src/features/worklog/types/WorkLog.ts
export interface WorkLog
{
    id: string;
    logNo: string;
    logDate: string;
    status:"DRAFT" | "SUBMITTED";
    summary: string | null;
    totalHours: number | null;
}

export interface WorkLogStats
{
    completionRate: number;
    draftCount: number;
    submittedCount: number;
}

export interface ApiResponse<T>
{
    success: boolean;
    data: T;
    message: string | null;
    errors: unknown[] | null;
}

export interface CreateWorkLog
{
    id: string;
    logNo: string;
    logDate: string;
    status: "DRAFT";
}

export interface ShiftType
{
    id: string;
    name: string;
    description: string | null;
}

export interface WorkItem
{
    id: string;
    seq: number;
    taskName: string;
    description: string | null;
    hours: number;
    progress: number;
}

export interface WorkItemRequest
{
    taskName:string;
    description: string | null;
    hours: number;
    progress: number;
}

export interface WorkItemMutationResponse
{
    item: WorkItem;
    logTotalHours: number;
}

export interface WorkLogDetail
{
    id: string;
    logNo: string;
    logDate: string;
    shiftType: ShiftType | null;
    status: "DRAFT" | "SUBMITTED";
    selfRead: boolean;
    totalHours: number | null;
    submittedAt: string | null;
    createdAt: string;
    updatedAt: string;
    workItems: WorkItem[];
}

export interface UpdateWorkLogRequest
{
    logDate:string;
    shiftTypeId:string;
    selfRead:boolean;
}

export interface SubmitWorkLogResponse
{
    id: string;
    logNo: string;
    status: "SUBMITTED";
    submittedAt: string;
}

File: WorkLog.API/src/Modules/WorkLogs/Modules.WorkLogs.Domains/WorkLogsAggregate/WorkLog.cs
using Modules.Common.Domains.SeedWork;

namespace Modules.WorkLogs.Domains.WorkLogsAggregate;

public class WorkLog : Entity, IAggregateRoot
{
    public WorkLog()
    {
        var now = DateTime.UtcNow;
        Id = Guid.NewGuid();
        CreatedAt = now;
        UpdatedAt = now;
    }
    public string LogNo { get; set; } = string.Empty;
    public Guid EmployeeId { get; set; }
    public DateOnly LogDate { get; set; }
    public Guid? ShiftTypeId { get; set; }
    public decimal? TotalHours { get; set; }
    public string Status { get; set; } = "DRAFT";
    public bool SelfRead { get; set; }
    public DateTime? SubmittedAt { get; set; }
    public List<WorkItem> WorkItems { get; set; } = [];
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class WorkItem
{
    public Guid Id { get; set; }
    public short Seq { get; set; }
    public string TaskName { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Hours { get; set; }
    public short? Progress { get; set; }
}